lib/dentaku/ast/identifier.rb

```diff
@@ -12,6 +12,8 @@ def initialize(token)
       def value(context={})
         v = context[identifier]
```

Comment:
Never mind, I don't see a way to avoid that behavior while keeping the recursive algorithm. So the behavior there always sets a key on the context. 

It seems to me that one path forward is to build in checks for nullity into all the relevant operators, so that you would throw the unbound variable error when performing a `+`, `-`, `/` etc.  That seems like a pain and complicates things, but I think it might have some benefits, like allowing an `IF` statement where the right side is irrelevant so it doesn't matter if its variable was passed or not. I won't work on that unless you think that is a good idea though.